test/autorest/generated/lrogroup/pollers.go

```diff
@@ -212,6 +231,10 @@ func (p *productPoller) Poll(ctx context.Context) (*http.Response, error) {
 }
 
 func (p *productPoller) FinalResponse(ctx context.Context) (*ProductResponse, error) {
+	product := Product{}
```

Comment:
If `product` isn't used elsewhere you can omit this definition and just use `Product{}` on the next line.

Reply:
The compiler doesnt like that, there's a problem with the braces
if ... Product{} {
 ...
}

Reply:
Ah, looks like parens are needed around the expression `(*p.resp.Product != Product{})`

Reply:
Ah ok, will do then

Reply:
Done